ols/src/query_helpers/query_docs.py

```diff
@@ -0,0 +1,124 @@
+"""This module has the components to retrieve docs from vector store."""
+from typing import Any, List
+
+from langchain_core.documents.base import Document
+from langchain_core.vectorstores import VectorStore, VectorStoreRetriever
+
+from ols.utils.logger import Logger
+
+
+class QueryDocs:
+    """Generic interface for retrieving docs from any vector store."""
+
+    def __init__(self) -> None:
+        """Initialize the `QueryDocs` instance."""
+        self.logger = Logger("QueryDocs").logger
+
+    def get_relevant_docs(
+        self,
+        query: str,
+        perform_query: bool = True,
+        vectordb: VectorStore = None,
+        search_type: str = "similarity",
+        **search_kwargs: Any,
+    ) -> List[Document] | None:
+        """Return list of documents from the vectorstore.
+
+        Args:
+            query: string to find relevant documents for
+            perform_query: Whether RAG retrieval needs to be done or not.
+            vectordb: Vector db from which you need to retrieve the docs.
+            search_type (Optional[str]): Defines the type of search that
+                the retriever should perform.
+                Can be "similarity" (default), "mmr", or
+                "similarity_score_threshold".
+            search_kwargs (Optional[Dict]): Keyword arguments to pass to the
+                search function. Can include things like:
+                    k: Amount of documents to return (Default: 4)
+                    score_threshold: Minimum relevance threshold
+                        for similarity_score_threshold
+                    fetch_k: Amount of documents to pass to MMR algorithm (
+                    Default: 20)
+                    lambda_mult: Diversity of results returned by MMR;
+                        1 for minimum diversity and 0 for maximum. (Default:
+                        0.5)
+                    filter: Filter by document metadata
+
+        Returns:
+            List[Document]: Returns a list of docs.
+
+        Examples:
+        .. code-block:: python
```

Comment:
This format `.. code-block:: python` seems more specific for sphinx style. Google's style should support 
```
Examples:
   ```python
   some code
   ```.
```
(ignore the dot, I'm fighting with formatting there)

But I guess this could be influence by the actual tool we choose for documentation build (IRC we don't know at the moment).

Reply:
done